### PR TITLE
Scripting: enforce GPLv2 for parsed stdlib docs

### PR DESCRIPTION
**Pull request by stu-elastic**
When parsing Java standard library javadocs, we need to ensure
that our use will comply with the license.  Our use complies
with GPLv2 licenses but may not comply with proprietary licenses.

Reject .java files that have non-GPL licenses when parsing them
for parameter names and javadoc comments.

### CODE REVIEW

File: modules/lang-painless/src/doc/java/org/elasticsearch/painless/ContextApiSpecGenerator.java [hunk 1]
```diff
@@ -70,11 +72,40 @@ private static Path resetRootDir() throws IOException {
...
+    public static class JavaClassFilesystemResolver implements JavaClassResolver {
+        private final Path root;
+
+        public JavaClassFilesystemResolver(Path root) {
+            this.root = root;
+        }
+
+        @SuppressForbidden(reason = "resolve class file from java src directory with environment")
+        public InputStream openClassFile(String className) throws IOException {
+            // TODO(stu): handle primitives & not stdlib
```
Comment (jdconrad) [inline modules/lang-painless/src/doc/java/org/elasticsearch/painless/ContextApiSpecGenerator.java:92]: Don't forget about the todo here. Maybe move to an issue?

**Review by jdconrad — APPROVED** [review]
LGTM! One minor comment.

File: modules/lang-painless/src/doc/java/org/elasticsearch/painless/ContextApiSpecGenerator.java (hunk 1 above)
Comment (stu-elastic) [reply modules/lang-painless/src/doc/java/org/elasticsearch/painless/ContextApiSpecGenerator.java:92]: Added https://github.com/elastic/elasticsearch/issues/68688

**Comment by stu-elastic** [comment]
Backport: 1152475
7.x (7.12): 3e89a66

**Pull request merged**